src/idm/group.js

```diff
@@ -0,0 +1,125 @@
+import axios from 'axios';
+import { errorHandler } from '../utils';
+
+const group = (base, token) => {
+  base = `${base}`;
+  return ({
+    /**
+    * @summary List all groups
+    * @name getGroups
+    * @public
+    * @function
+    * @memberof agile.idm.group
+    * @fulfil {Array} all groups
+    * @returns {Promise}
+    * @example
+    * agile.idm.group.getGroups().then(function(groups) {
+    *   console.log(groups);
+    * });
+    **/
+    getGroups: () => {
+      let url = `${base}/api/v1/group`;
+      return axios({
+        method: 'GET',
+        url: url,
+        headers: {"Authorization":`bearer ${token}`}
+      })
+      .then(res => (res.data))
+      .catch(errorHandler);
+    },
+    /**
+    * @summary Create a group onwned by the authenticated user
+    * @name createGroup
+    * @public
+    * @function
+    * @memberof agile.idm.group
+    * @param {String} groupName - Name of the group
+    * @fulfil {Object} group created
+    * @returns {Promise}
+    * @example
+    * agile.idm.group.createGroup('ble-devices').then(function(group) {
+    *   console.log('group created!'+group);
+    * });
+    **/
+    createGroup: (name) => axios({
+      method: 'POST',
+      url: `${base}/api/v1/group/`,
+      headers: {"Authorization" : `bearer ${token}`},
```

Comment:
given that we may not need the interceptor always, e.g. client authentication, I created instances inside each idm namespace instead of setting it globally (some additional info here https://github.com/mzabriskie/axios/issues/108)